packages/react/src/hooks/useDataStore.tsx

```diff
@@ -11,99 +11,107 @@ import {
 } from '../primitives/types/datastore';
 
 /**
- * Perform a collection query against a DataStore model
+ * Perform a query against a DataStore model
+ * @param props object containing params for calling either calling Datastore.query or Datastore.observeQuery
+ * @returns the result of either calling Datastore.query or Datastore.observeQuery for the provided props
  * @internal
  */
-export const useDataStoreCollection = <M extends PersistentModel>({
-  model,
-  criteria,
-  pagination,
-}: DataStoreCollectionProps<M>): DataStoreCollectionResult<M> => {
-  const [result, setResult] = React.useState<DataStoreCollectionResult<M>>({
-    items: [],
-    isLoading: false,
-    error: undefined,
-  });
+export function useDataStoreBinding<Model extends PersistentModel>(
+  props: DataStoreBindingProps<Model, 'record'>
+): DataStoreItemResult<Model>;
+export function useDataStoreBinding<Model extends PersistentModel>(
+  props: DataStoreBindingProps<Model, 'collection'>
+): DataStoreCollectionResult<Model>;
+export function useDataStoreBinding<Model extends PersistentModel>(
+  props:
+    | DataStoreBindingProps<Model, 'record'>
+    | DataStoreBindingProps<Model, 'collection'>
+): DataStoreItemResult<Model> | DataStoreCollectionResult<Model> {
```

Comment:
I'm sad that we did this, but it looks like codegen-ui is dependent on this behavior rather than just the underlying hooks.